gammapy/spectrum/dataset.py

```diff
@@ -343,6 +338,23 @@ def likelihood_per_bin(self):
         )
         return np.nan_to_num(on_stat_)
 
+    def fake(self, random_state="random-seed"):
+        """Simulate fake data for the counts from a Poisson Law of mean value = N predicted counts in each energy bin.
+        Simulate fake data for the background from a Poisson Law of mean value = N background counts in each energy bin.
+        Erase the counts and background data and replace them by the simulated one.
+
+        Parameters
+        ----------
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+                Defines random number generator initialisation.
+                Passed to `~gammapy.utils.random.get_random_state`.
+        """
+        random_state = get_random_state(random_state)
```

Comment:
You should test that `self.background is not None` and possibly raise an exception if it is not there.